src/main/java/duke/logic/Parser.java

```diff
@@ -0,0 +1,275 @@
+package duke.logic;
+
+import duke.command.*;
+import duke.exception.*;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Locale;
+
+/**
+ * Parser deals with making sense of user input.
+ *
+ * @author totsukatomofumi
+ */
+public class Parser {
+    /** Task list to be passed into command constructors. */
+    private TaskList taskList;
+
+    /** HashMap to contain abbreviated month and their numerical representation as a string pairs. */
+    private static HashMap<String, String> months = new HashMap<>();
+
+    static {
+        String[] monthsArr = {"jan", "feb", "mar", "apr", "may", "jun", "jul", "aug", "sep", "oct", "nov", "dec"};
```

Comment:
Since the string array is named `monthsArr`, would it better for the hashmap to be named `monthsMap`?